From 774bfefb613a465981d4973b654deaf2d750f054 Mon Sep 17 00:00:00 2001
From: Qi Liu <liuqi115@huawei.com>
Date: Wed, 30 Nov 2022 19:01:16 +0800
Subject: [PATCH 1220/1646] [Backport] perf pmu: Add alias match method to fit
 pmu_name of HiSilicon DDRC

driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I63VF5

Reference: https://gitee.com/openeuler/kernel/commit/fd292e0824e8453730e8ad039cc920f802562bea

--------------------------------

pmu_name of DDRC PMU on Hip09 platform is set as "hisi_sllcX_ddrcY_Z",
current match method of pmu_name cannot fit DDRC pmu, so modify the method.

Signed-off-by: Qi Liu <liuqi115@huawei.com>
Signed-off-by: Junhao He <hejunhao3@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
---
 tools/perf/util/pmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/perf/util/pmu.c b/tools/perf/util/pmu.c
index e76be445512b..cafc842f8ada 100644
--- a/tools/perf/util/pmu.c
+++ b/tools/perf/util/pmu.c
@@ -739,7 +739,7 @@ static bool perf_pmu__valid_suffix(const char *pmu_name, char *tok)
 
 	/* Ensure we end in a number */
 	while (1) {
-		if (!isdigit(*p))
+		if (!isdigit(*p) && (*p != '_'))
 			return false;
 		if (*(++p) == 0)
 			break;
-- 
2.17.1

